src/v2/services/implementations/DappStakingService.ts

```diff
@@ -0,0 +1,35 @@
+import { ethers } from 'ethers';
+import { inject, injectable } from 'inversify-props';
+import { Guard } from 'src/v2/common';
+import { TvlModel } from 'src/v2/models';
+import { IDappStakingRepository, IMetadataRepository, IPriceRepository } from 'src/v2/repositories';
+import { Symbols } from 'src/v2/symbols';
+import { IDappStakingService } from 'src/v2/services';
+
+@injectable()
+export class DappStakingService implements IDappStakingService {
+  constructor(
+    @inject() private dappStakingRepository: IDappStakingRepository,
+    @inject(Symbols.CoinGecko) private priceRepository: IPriceRepository,
+    @inject() private metadataRepository: IMetadataRepository
+  ) {
+    Guard.ThrowIfUndefined('dappStakingRepository', dappStakingRepository);
```

Comment:
Overall codes are great. but just curious that do we need to always put this kind of function to check undefined for all injectable params? it seems boilerplate a bit.

Reply:
@sirius651 Thank you for the review. I believe you are right. We don't need to check injected parameters since TypeScript wont' allow undefined values there. I will update that. On the other hand we need to check function parameters (e.g. strings) which will help us to notice/prevent bugs earlier.

Reply:
Fixed in commit https://github.com/AstarNetwork/astar-apps/pull/435/commits/8ae4688b257b28deb7f37958713e6b6d9920da53